### ISSUE TITLE
Crash caused by setting the plot biome

### ISSUE THREAD

**Issue by SecretlyJealous** [IntellectualSites/FastAsyncWorldEdit#670]
# Bug report for FastAsyncWorldEdit 1.16.3

**[REQUIRED] FastAsyncWorldEdit Configuration Files:** 
config.yml: https://pastebin.com/qbx642wn
config-legacy.yml: https://pastebin.com/E4qmuBMb

**[REQUIRED] FastAsyncWorldEdit Version Number:** 
FastAsyncWorldEdit-1.16-376

**[REQUIRED] Spigot/Paper Version Number:** 
git-Paper-204 (MC: 1.16.3) (Implementing API version 1.16.3-R0.1-SNAPSHOT)

**[REQUIRED] Description of the problem:**
From the crash log it looks like someone did a **/p setbiome plains** which caused the server to freeze and crash.
Crash log: https://pastebin.com/S62m4Dvv

Within the crash log it shows the server is stuck waiting on this:
```
[21:05:49] [Paper Watchdog Thread/ERROR]: ------------------------------
[21:05:49] [Paper Watchdog Thread/ERROR]: Current Thread: Craft Scheduler Thread - 724
[21:05:49] [Paper Watchdog Thread/ERROR]:   PID: 12115 | Suspended: false | Native: false | State: WAITING
[21:05:49] [Paper Watchdog Thread/ERROR]:   Thread is waiting on monitor(s):
[21:05:49] [Paper Watchdog Thread/ERROR]:       Locked on:com.boydti.fawe.bukkit.adapter.mc1_16_2.BukkitGetBlocks_1_16_2.call(BukkitGetBlocks_1_16_2.java:546)
[21:05:49] [Paper Watchdog Thread/ERROR]:       Locked on:com.boydti.fawe.beta.implementation.chunk.ChunkHolder.call(ChunkHolder.java:771)
[21:05:49] [Paper Watchdog Thread/ERROR]:       Locked on:com.boydti.fawe.beta.implementation.queue.SingleThreadQueueExtent.flush(SingleThreadQueueExtent.java:324)
[21:05:49] [Paper Watchdog Thread/ERROR]:       Locked on:com.boydti.fawe.bukkit.regions.plotsquared.FaweRegionManager.lambda$setBiome$3(FaweRegionManager.java:219)
[21:05:49] [Paper Watchdog Thread/ERROR]:   Stack:
[21:05:49] [Paper Watchdog Thread/ERROR]:       sun.misc.Unsafe.park(Native Method)
[21:05:49] [Paper Watchdog Thread/ERROR]:       java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
[21:05:49] [Paper Watchdog Thread/ERROR]:       java.util.concurrent.FutureTask.awaitDone(FutureTask.java:429)
[21:05:49] [Paper Watchdog Thread/ERROR]:       java.util.concurrent.FutureTask.get(FutureTask.java:191)
[21:05:49] [Paper Watchdog Thread/ERROR]:       com.boydti.fawe.util.TaskManager.sync(TaskManager.java:336)
[21:05:49] [Paper Watchdog Thread/ERROR]:       com.boydti.fawe.bukkit.adapter.mc1_16_2.BukkitAdapter_1_16_2.sendChunk(BukkitAdapter_1_16_2.java:192)
[21:05:49] [Paper Watchdog Thread/ERROR]:       com.boydti.fawe.bukkit.adapter.mc1_16_2.BukkitGetBlocks_1_16_2.lambda$call$7(BukkitGetBlocks_1_16_2.java:513)
[21:05:49] [Paper Watchdog Thread/ERROR]:       com.boydti.fawe.bukkit.adapter.mc1_16_2.BukkitGetBlocks_1_16_2$$Lambda$6152/899039346.run(Unknown Source)
[21:05:49] [Paper Watchdog Thread/ERROR]:       com.boydti.fawe.bukkit.adapter.mc1_16_2.BukkitGetBlocks_1_16_2.call(BukkitGetBlocks_1_16_2.java:546)
[21:05:49] [Paper Watchdog Thread/ERROR]:       com.boydti.fawe.beta.implementation.chunk.ChunkHolder.call(ChunkHolder.java:783)
[21:05:49] [Paper Watchdog Thread/ERROR]:       com.boydti.fawe.beta.implementation.chunk.ChunkHolder.call(ChunkHolder.java:771)
[21:05:49] [Paper Watchdog Thread/ERROR]:       com.boydti.fawe.beta.implementation.chunk.ChunkHolder.call(ChunkHolder.java:32)
[21:05:49] [Paper Watchdog Thread/ERROR]:       java.util.concurrent.FutureTask.run(FutureTask.java:266)
[21:05:49] [Paper Watchdog Thread/ERROR]:       java.util.concurrent.ThreadPoolExecutor$CallerRunsPolicy.rejectedExecution(ThreadPoolExecutor.java:2038)
[21:05:49] [Paper Watchdog Thread/ERROR]:       java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:830)
[21:05:49] [Paper Watchdog Thread/ERROR]:       java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1379)
[21:05:49] [Paper Watchdog Thread/ERROR]:       java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:134)
[21:05:49] [Paper Watchdog Thread/ERROR]:       com.boydti.fawe.beta.implementation.queue.QueueHandler.submit(QueueHandler.java:236)
[21:05:49] [Paper Watchdog Thread/ERROR]:       com.boydti.fawe.beta.implementation.queue.SingleThreadQueueExtent.submitUnchecked(SingleThreadQueueExtent.java:180)
[21:05:49] [Paper Watchdog Thread/ERROR]:       com.boydti.fawe.beta.implementation.queue.SingleThreadQueueExtent.flush(SingleThreadQueueExtent.java:324)
[21:05:49] [Paper Watchdog Thread/ERROR]:       com.boydti.fawe.beta.IQueueExtent.commit(IQueueExtent.java:113)
[21:05:49] [Paper Watchdog Thread/ERROR]:       com.sk89q.worldedit.extent.PassthroughExtent.commit(PassthroughExtent.java:209)
[21:05:49] [Paper Watchdog Thread/ERROR]:       com.sk89q.worldedit.extent.PassthroughExtent.commit(PassthroughExtent.java:209)
[21:05:49] [Paper Watchdog Thread/ERROR]:       com.sk89q.worldedit.EditSession.flushQueue(EditSession.java:1044)
[21:05:49] [Paper Watchdog Thread/ERROR]:       com.boydti.fawe.bukkit.regions.plotsquared.FaweRegionManager.lambda$setBiome$3(FaweRegionManager.java:219)
[21:05:49] [Paper Watchdog Thread/ERROR]:       com.boydti.fawe.bukkit.regions.plotsquared.FaweRegionManager$$Lambda$6436/799585937.run(Unknown Source)
[21:05:49] [Paper Watchdog Thread/ERROR]:       org.bukkit.craftbukkit.v1_16_R2.scheduler.CraftTask.run(CraftTask.java:99)
[21:05:49] [Paper Watchdog Thread/ERROR]:       org.bukkit.craftbukkit.v1_16_R2.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:54)
[21:05:49] [Paper Watchdog Thread/ERROR]:       com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22)
[21:05:49] [Paper Watchdog Thread/ERROR]:       java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[21:05:49] [Paper Watchdog Thread/ERROR]:       java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[21:05:49] [Paper Watchdog Thread/ERROR]:       java.lang.Thread.run(Thread.java:748)
[21:05:49] [Paper Watchdog Thread/ERROR]: ------------------------------
```
This crash also looks similar to the crash described by another user in this ticket: https://github.com/IntellectualSites/FastAsyncWorldEdit/issues/471

Crash log: https://pastebin.com/W7PMG8Zy
Within this user's crash log it shows the server is stuck waiting on this:

```
[13:46:10] [Paper Watchdog Thread/ERROR]: ------------------------------
[13:46:10] [Paper Watchdog Thread/ERROR]: Current Thread: Craft Scheduler Thread - 1046
[13:46:10] [Paper Watchdog Thread/ERROR]: 	PID: 352987 | Suspended: false | Native: false | State: WAITING
[13:46:10] [Paper Watchdog Thread/ERROR]: 	Thread is waiting on monitor(s):
[13:46:10] [Paper Watchdog Thread/ERROR]: 		Locked on:com.boydti.fawe.bukkit.adapter.mc1_15_2.BukkitGetBlocks_1_15_2.getSections(BukkitGetBlocks_1_15_2.java:623)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		Locked on:com.boydti.fawe.beta.implementation.chunk.ChunkHolder.call(ChunkHolder.java:424)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		Locked on:com.boydti.fawe.beta.implementation.queue.SingleThreadQueueExtent.flush(SingleThreadQueueExtent.java:323)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		Locked on:com.boydti.fawe.bukkit.regions.plotsquared.FaweRegionManager.lambda$setBiome$1(FaweRegionManager.java:82)
[13:46:10] [Paper Watchdog Thread/ERROR]: 	Stack:
[13:46:10] [Paper Watchdog Thread/ERROR]: 		java.base@11.0.6/jdk.internal.misc.Unsafe.park(Native Method)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		java.base@11.0.6/java.util.concurrent.locks.LockSupport.park(LockSupport.java:194)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		java.base@11.0.6/java.util.concurrent.CompletableFuture$Signaller.block(CompletableFuture.java:1796)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		java.base@11.0.6/java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3128)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		java.base@11.0.6/java.util.concurrent.CompletableFuture.waitingGet(CompletableFuture.java:1823)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		java.base@11.0.6/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1998)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		com.boydti.fawe.bukkit.adapter.mc1_15_2.BukkitAdapter_1_15_2.ensureLoaded(BukkitAdapter_1_15_2.java:138)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		com.boydti.fawe.bukkit.adapter.mc1_15_2.BukkitGetBlocks_1_15_2.ensureLoaded(BukkitGetBlocks_1_15_2.java:239)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		com.boydti.fawe.bukkit.adapter.mc1_15_2.BukkitGetBlocks_1_15_2.getChunk(BukkitGetBlocks_1_15_2.java:637)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		com.boydti.fawe.bukkit.adapter.mc1_15_2.BukkitGetBlocks_1_15_2.getSections(BukkitGetBlocks_1_15_2.java:623)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		com.boydti.fawe.bukkit.adapter.mc1_15_2.BukkitGetBlocks_1_15_2.hasSection(BukkitGetBlocks_1_15_2.java:646)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		com.boydti.fawe.bukkit.adapter.mc1_15_2.BukkitGetBlocks_1_15_2.trim(BukkitGetBlocks_1_15_2.java:657)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		com.boydti.fawe.beta.implementation.chunk.ChunkHolder.call(ChunkHolder.java:433)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		com.boydti.fawe.beta.implementation.chunk.ChunkHolder.call(ChunkHolder.java:424)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		com.boydti.fawe.beta.implementation.chunk.ChunkHolder.call(ChunkHolder.java:31)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		java.base@11.0.6/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		java.base@11.0.6/java.util.concurrent.ThreadPoolExecutor$CallerRunsPolicy.rejectedExecution(ThreadPoolExecutor.java:2027)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		java.base@11.0.6/java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:825)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		java.base@11.0.6/java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1355)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		java.base@11.0.6/java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:140)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		com.boydti.fawe.beta.implementation.queue.QueueHandler.submit(QueueHandler.java:239)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		com.boydti.fawe.beta.implementation.queue.SingleThreadQueueExtent.submitUnchecked(SingleThreadQueueExtent.java:179)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		com.boydti.fawe.beta.implementation.queue.SingleThreadQueueExtent.flush(SingleThreadQueueExtent.java:323)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		com.boydti.fawe.beta.IQueueExtent.commit(IQueueExtent.java:112)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		com.sk89q.worldedit.extent.PassthroughExtent.commit(PassthroughExtent.java:213)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		com.sk89q.worldedit.extent.PassthroughExtent.commit(PassthroughExtent.java:213)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		com.sk89q.worldedit.EditSession.flushQueue(EditSession.java:1013)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		com.boydti.fawe.bukkit.regions.plotsquared.FaweRegionManager.lambda$setBiome$1(FaweRegionManager.java:82)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		com.boydti.fawe.bukkit.regions.plotsquared.FaweRegionManager$$Lambda$6685/0x0000000841a87c40.run(Unknown Source)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		app//org.bukkit.craftbukkit.v1_15_R1.scheduler.CraftTask.run(CraftTask.java:84)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		app//org.bukkit.craftbukkit.v1_15_R1.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:54)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		app//com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		java.base@11.0.6/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		java.base@11.0.6/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[13:46:10] [Paper Watchdog Thread/ERROR]: 		java.base@11.0.6/java.lang.Thread.run(Thread.java:834)
[13:46:10] [Paper Watchdog Thread/ERROR]: ------------------------------
```

**How to replicate**:
I'm not quite sure, this seems like some sort of edge case scenario

**Plugins being used on the server:**
FastAsyncWorldEdit, LuckPerms, PlotSquared, ProtocolLib, LibsDisguises, SimplePets

**Checklist**:

- [X] I included all information required in the sections above
- [X] I made sure there are no duplicates of this report [(Use Search)](https://github.com/IntellectualSites/FastAsyncWorldEdit/issues?q=is%3Aissue)
- [X] I made sure I am using an up-to-date version of [FastAsyncWorldEdit for 1.16.3](https://ci.athion.net/job/FastAsyncWorldEdit-1.16/)
- [X] I made sure the bug/error is not caused by any other plugin

**Comment by dordsor21** [comment]
You have a number if mismatching error logs. Some are 1.16, some are 1.15, with the important one (the full crash) being 1.15. Please adjust your information to be useful.
